src/ai/backend/web/server.py

```diff
@@ -654,6 +654,18 @@ async def server_main(
     anon_web_handler = partial(web_handler, is_anonymous=True)
     anon_web_plugin_handler = partial(web_plugin_handler, is_anonymous=True)
 
+    pipeline_api_endpoint = config["pipeline"]["endpoint"]
+    pipeline_handler = partial(web_handler, is_anonymous=True, api_endpoint=pipeline_api_endpoint)
+    pipeline_login_handler = partial(
+        web_handler,
+        is_anonymous=False,
+        api_endpoint=pipeline_api_endpoint,
+        extra_forwarding_headers={"X-BackendAI-SessionID"},
+    )
+    pipeline_websocket_handler = partial(
+        websocket_handler, is_anonymous=True, api_endpoint=pipeline_api_endpoint.with_scheme("ws")
+    )
```

Comment:
It might be cleaner to separate and pass it from the handler layer. Please create a follow-up issue for refactoring.